### PR TITLE
Sh/dependency separation and clean up

### PR DESCRIPTION
**Pull request by sherholz**
Separating the USD dependency libraries into separate `cmake\X.cmake` files.
Added options to build all dependencies or using the ones installed on the system.
Added support to build ColorIO, OpenVDB and PRman plugins.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,169 +40,192 @@ include(GNUInstallDirs)
...
+option(BUILD_VDB "build OpenVDB"              OFF)
+option(BUILD_BOOST "build BOOST"              OFF)
+option(BUILD_OPENEXR "build OpenEXR"          OFF)
+option(BUILD_OPENIMAGEIO "build OpenImageIO"  OFF)
+option(BUILD_OPENCOLORIO "build OpenColorIO"  OFF)
+option(BUILD_PTEX        "build Ptex"          ON)
+option(BUILD_OPENSUBDIV  "build OpenSubdiv"    ON)
+
+if(BUILD_ALL_DEPS)
+#  set(BUILD_GLEW ON)
```
Comment (carsonbrownlee) [inline CMakeLists.txt:65]: ?

File: cmake/Boost.cmake [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  set(BOOST_BUILD_COMMAND ${BOOST_BUILD_COMMAND}  --user-config=${CMAKE_CURRENT_BINARY_DIR}/source/boost/python-config.jam python=${BOOST_PYTHON_VERSIONS} --with-python)
+
+endif()
+
+ExternalProject_Add (
+	${EP_BOOST}
+
+	PREFIX         ${EP_BOOST}/source/boost
+	BUILD_IN_SOURCE 1
+	URL https://downloads.sourceforge.net/project/boost/boost/1.67.0/boost_1_67_0.tar.gz
```
Comment (carsonbrownlee) [inline cmake/Boost.cmake:42]: why did you change the boost version?  Note that I had it split between version 1.61 and 1.70 for python2 vs python3 as I ran into issues previously with python3 using the older boost and vise versa.  Did you try the build with both python3 and python2 with this version?  I tried it with python2 and 1.67 and it does work on my ubuntu build, though I'm not sure about python3.

**Comment by carsonbrownlee** [comment]
@sherholz looks good and works on ubuntu 10.04 with python2, but added some comments.  Overall there is a lot of commented out code.  Can that be removed before merging?

File: cmake/Boost.cmake (hunk 2 above)
Comment (sherholz) [reply cmake/Boost.cmake:42]: This version was the only one of the three where I was able to build 'boost_python' for Python2 and Python3 at the same time without any problems.

File: CMakeLists.txt (hunk 1 above)
Comment (sherholz) [reply CMakeLists.txt:65]: This way you are able to either build all dependencies or only a subset and use the system-wide installed versions for the others.

Comment (sherholz) [reply CMakeLists.txt:65]: This way you are able to either build all dependencies or only a subset and use the system-wide installed versions for the others.

File: cmake/Boost.cmake (hunk 2 above)
Comment (carsonbrownlee) [reply cmake/Boost.cmake:42]: ok great, I had to play version roulette to find one that worked for each.

File: CMakeLists.txt (hunk 1 above)
Comment (carsonbrownlee) [reply CMakeLists.txt:65]: no, I meant why is build_glew the only dependency commented out?
